src/NServiceBus.Core/Sagas/SagaPersistenceBehavior.cs

```diff
@@ -329,8 +331,16 @@ IContainSagaData CreateNewSagaEntity(SagaMetadata metadata, IInvokeHandlerContex
                     .ConvertFromInvariantString(value.PropertyValue.ToString());
 
                 propertyInfo.SetValue(sagaEntity, convertedValue);
+
+                sagaIdGeneratorContext = new SagaIdGeneratorContext(value.PropertyName, value.PropertyValue, metadata, context.Extensions);
+            }
+            else
+            {
+                sagaIdGeneratorContext = new SagaIdGeneratorContext(null, null, metadata, context.Extensions);
             }
 
+            sagaEntity.Id = sagaIdGenerator.Generate(sagaIdGeneratorContext);
```

Comment:
This could be simplified by doing this instead:

```csharp
var sagaIdGeneratorContext = new SagaIdGeneratorContext(value?.PropertyName, value?.PropertyValue, metadata, context.Extensions);
```